java/main/src/test/java/com/saucelabs/saucebindings/options/VisualOptionsTest.java

```diff
@@ -0,0 +1,269 @@
+        assertEquals(Duration.ofSeconds(59), sauceOptions.getPageLoadTimeout());
+        assertEquals(Duration.ofSeconds(29), sauceOptions.getScriptTimeout());
+        assertEquals(true, sauceOptions.sauce().getAvoidProxy());
+        assertEquals("Sample Build Name", sauceOptions.sauce().getBuild());
+        assertEquals(true, sauceOptions.sauce().getCapturePerformance());
+        assertEquals("71", sauceOptions.sauce().getChromedriverVersion());
+        assertEquals(Integer.valueOf(2), sauceOptions.sauce().getCommandTimeout());
+        assertEquals(customData, sauceOptions.sauce().getCustomData());
+        assertEquals(true, sauceOptions.sauce().getExtendedDebugging());
+        assertEquals(Integer.valueOf(3), sauceOptions.sauce().getIdleTimeout());
+        assertEquals("3.141.0", sauceOptions.sauce().getIedriverVersion());
+        assertEquals(Integer.valueOf(300), sauceOptions.sauce().getMaxDuration());
+        assertEquals("Sample Test Name", sauceOptions.sauce().getName());
+        assertEquals("Mommy", sauceOptions.sauce().getParentTunnel());
+        assertEquals(prerun, sauceOptions.sauce().getPrerun());
+        assertEquals(Integer.valueOf(0), sauceOptions.sauce().getPriority());
+        assertEquals(JobVisibility.TEAM, sauceOptions.sauce().getJobVisibility());
+        assertEquals(false, sauceOptions.sauce().getRecordLogs());
+        assertEquals(false, sauceOptions.sauce().getRecordScreenshots());
+        assertEquals(false, sauceOptions.sauce().getRecordVideo());
+        assertEquals("10x10", sauceOptions.sauce().getScreenResolution());
+        assertEquals("3.141.59", sauceOptions.sauce().getSeleniumVersion());
+        assertEquals(tags, sauceOptions.sauce().getTags());
+        assertEquals("San Francisco", sauceOptions.sauce().getTimeZone());
+        assertEquals("tunnelname", sauceOptions.sauce().getTunnelIdentifier());
+        assertEquals(false, sauceOptions.sauce().getVideoUploadOnPass());
+
+        assertEquals("My Project", visualOptions.getProjectName());
+        assertEquals("393x786", visualOptions.getViewportSize());
+        assertEquals("baseBranch", visualOptions.getBaseBranch());
+        assertEquals(diffOptions, visualOptions.getDiffOptions());
+        assertEquals("#some-id, .some-selector", visualOptions.getIgnore());
+        assertTrue(visualOptions.getFailOnNewStates());
+        assertTrue(visualOptions.getAlwaysAcceptBaseBranch());
+        assertTrue(visualOptions.getDisableBranchBaseline());
+        assertTrue(visualOptions.getScrollAndStitchScreenshots());
+        assertTrue(visualOptions.getDisableCORS());
+        assertTrue(visualOptions.getIframes());
+        assertEquals(iframesOptions, visualOptions.getIframesOptions());
+    }
+
+    @SneakyThrows
+    public Map<String, Object> serialize(String key) {
+        InputStream input = new FileInputStream(new File("src/test/java/com/saucelabs/saucebindings/options.yml"));
```

Comment:
I usually use something like https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html with closeable streams to make sure they are properly closed